pi/heartbeat.py

```diff
@@ -207,4 +213,10 @@ def send_heartbeat(flic_last_seen_secs, flic_last_ping_secs, system_id):
                 else:
                     relay.off()
                     led.off()
-                time.sleep(10)
+
+                # run the loop at most once every 10 seconds
+                # if the run loop takes longer then 10 seconds we don't need to sleep (eg. if http requests time out)
+                run_loop_duration = (datetime.datetime.now() - run_loop_last_run_time).total_seconds()
+                sleep_time = max(10.0 - run_loop_duration, 0.0)
+                time.sleep(sleep_time)
```

Comment:
What do you think of factoring this out into a function and then adding testing?